### PR TITLE
Executor version environment to use proper version of Runtimes

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

Adds executor version environment variable (_APP_EXECUTOR_VERSION), currently using only internally that will allow us to use the correct version of runtimes depending on the executor version.

## Test Plan

Existing tests

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/config/runtimes.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 /**
  * List of Appwrite Cloud Functions supported runtimes
  */
-$runtimes = new Runtimes();
+$runtimes = new Runtimes(App::getEnv('_APP_EXECUTOR_VERSION'));
```
Comment (Meldiron) [inline app/config/runtimes.php:9]: Can we have a default here?

```suggestion
$runtimes = new Runtimes(App::getEnv('_APP_EXECUTOR_VERSION', 'v2'));
```

File: app/config/variables.php [hunk 2]
```diff
@@ -615,6 +614,15 @@
                 'question' => '',
                 'filter' => ''
             ],
+            [
+                'name' => '_APP_EXECUTOR_VERSION',
+                'description' => 'Executor version',
+                'introduction' => '0.14.0',
+                'default' => 'v1',
```
Comment (Meldiron) [inline app/config/variables.php:621]:
```suggestion
                'default' => 'v2',
```
In 0.14 we will use V2, right?

File: app/executor.php [hunk 3]
```diff
@@ -584,7 +584,7 @@ function (string $runtimeId, array $vars, string $data, $timeout, $activeRuntime
     /** 
      * Warmup: make sure images are ready to run fast 🚀
      */
-    $runtimes = new Runtimes();
+    $runtimes = new Runtimes(App::getEnv('_APP_EXECUTOR_VERSION'));
```
Comment (Meldiron) [inline app/executor.php:587]:
```suggestion
    $runtimes = new Runtimes(App::getEnv('_APP_EXECUTOR_VERSION', 'v2'));
```

Same as before

File: Dockerfile [hunk 4]
```diff
@@ -187,6 +187,7 @@ ENV _APP_SERVER=swoole \
     _APP_EXECUTOR_SECRET=a-random-secret \
     _APP_EXECUTOR_HOST=http://appwrite-executor/v1 \
     _APP_EXECUTOR_RUNTIME_NETWORK=appwrite_runtimes \
```
Comment (Meldiron) [inline Dockerfile:189]: I think we are missing:
-  value in `.env`
- binding in `docker-compose.yml`
- Both of the above for released files (they are in PHTMLs)

**Review by Meldiron — CHANGES_REQUESTED** [review]

File: app/config/variables.php (hunk 2 above)
Comment (lohanidamodar) [reply app/config/variables.php:621]: There was never V1, so probably we will use V1 still. As executor is still part of Appwrite, we don't have versioning here yet.

File: Dockerfile (hunk 4 above)
Comment (lohanidamodar) [reply Dockerfile:189]: As you can see my comment in the description, we are not yet documenting this Environment variable publicly, for now we will be using it internally.

**Review by Meldiron — APPROVED** [review]

**Comment by TorstenDittmann** [comment]
Why does this need a ENV variable?

Shouldn't this be part of the composer dependency?

**Pull request merged**